app/controllers/agents_controller.rb

```diff
@@ -111,8 +114,10 @@ def propagate
   end
 
   def destroy_memory
-    @agent = current_user.agents.find(params[:id])
-    @agent.update!(memory: {})
+    Agent.transaction do
+      @agent = current_user.agents.lock.find(params[:id])
+      @agent.update!(memory: {})
+    end
```

Comment:
Mostly just curious, in some places you used `.with_lock` and in other wrapped `.lock.find` in a transaction "manually", afaik they are interchangeable functionality wise. WDYT about using one pattern everywhere it's feasible? I generally think `with_lock` is easier to read

```suggestion
    @agent = current_user.agents.find(params[:id])
    @agent.with_lock do
      @agent.update!(memory: {})
    end
```

Reply:
That would cost you two select queries when you needed just one and could lead to a race condition.
```
SELECT; -- find
-- The record may change
BEGIN;  -- with_lock do
  SELECT FOR UPDATE;
  -- ...
COMMIT; -- end
```

Reply:
Of course, Huginn will never change `id` or the ownership of an agent so the race condition would do little to no harm in this particular case, but I'm talking about it in general.